packages/app/src/systems/Chains/fuel/utils/contract.ts

```diff
@@ -1,8 +1,11 @@
-import { ZeroBytes32, hash, concat } from 'fuels';
+import { concatBytes } from '@fuel-ts/utils';
+import { ZeroBytes32, arrayify, hash } from 'fuels';
 
 export function getContractTokenId(
   contractId: `0x${string}`,
   subId = ZeroBytes32
 ) {
-  return hash(concat([contractId, subId]));
+  const byteContract = arrayify(contractId);
+  const byteSubId = arrayify(subId);
+  return hash(concatBytes([byteContract, byteSubId]));
```

Comment:
What is the difference between the contact from ether with the contact from fuel?

Both should just contact the array of bytes.

Reply:
stuff changed in fuels-ts and I created this PR https://github.com/FuelLabs/fuels-ts/pull/1365